src/Filter/StringFilter.php

```diff
@@ -16,13 +16,14 @@
 use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
 use Sonata\AdminBundle\Form\Type\Filter\ChoiceType;
 use Sonata\AdminBundle\Form\Type\Operator\ContainsOperatorType;
+use Sonata\AdminBundle\Form\Type\Operator\StringOperatorType;
 
 class StringFilter extends Filter
 {
     public const CHOICES = [
         ContainsOperatorType::TYPE_CONTAINS => 'LIKE',
-        ContainsOperatorType::TYPE_STARTS_WITH => 'LIKE',
-        ContainsOperatorType::TYPE_ENDS_WITH => 'LIKE',
+        StringOperatorType::TYPE_STARTS_WITH => 'LIKE',
+        StringOperatorType::TYPE_ENDS_WITH => 'LIKE',
         ContainsOperatorType::TYPE_NOT_CONTAINS => 'NOT LIKE',
         ContainsOperatorType::TYPE_EQUAL => '=',
```

Comment:
Dont mix ContainsOperatorType and StringOperatorType.

I was expecting you to add the 5 constant in the StringOperatorType.

Then you only works with the StringOperatorType constant

You can then configure the `public function getRenderSettings()` to use the StringOperatorType

'operator_type' => StringOperatorType::class,

Reply:
Let me try.

Reply:
The last needed thing IMHO would be to update the current tests and add one for start with and one for end with.

Ping me if you need help

Reply:
I have never written tests. So I have no idea. I will try to do. but not sure.

Reply:
All your tests are and will be here https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/tests/Filter/StringFilterTest.php

First you need to change the ContainsOperatorType for StringOperatorType.

Then you can basically copy paste the test
```
public function testContains()
```

And change the name, and the StringOperatorType:: const used in order to create the testStartWith and testEndWith.

The only difference will certainly be
```
$this->assertSame(['field_name_0' => '%asd'], $builder->parameters);
```
or
```
$this->assertSame(['field_name_0' => 'asd%'], $builder->parameters);
```

Reply:
Added. Please have a look. Thanks